lib/presentation/profile/widget/profile_picture.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+
+class ProfilePicture extends StatelessWidget {
+  const ProfilePicture({
+    Key? key,
+    File? image,
+    String? userId,
+    required double maxRadius,
+  })  : _image = image,
+        _userId = userId,
+        _maxRadius = maxRadius,
+        super(key: key);
+
+  final File? _image;
+  final String? _userId;
+  final double _maxRadius;
+
+  @override
+  Widget build(BuildContext context) {
+    ImageProvider? imageProvider;
+    if (_image != null) {
+      // ignore: unnecessary_cast
+      imageProvider = FileImage(_image!) as ImageProvider<Object>;
+    } else if (_userId != null) {
+      imageProvider = NetworkImage(
+        'https://static-dev.collaction.org/profile-pictures/$_userId.png',
+      );
+    }
+
+    return CircleAvatar(
+      maxRadius: _maxRadius,
+      foregroundImage: imageProvider,
+      backgroundImage: const AssetImage(
+        'assets/images/logo.png',
+      ),
+    );
+  }
```

Comment:
```suggestion
  Widget build(BuildContext context) {
    return CircleAvatar(
      maxRadius: _maxRadius,
      foregroundImage: _image != null ? FileImage(_image!) as ImageProvider<Object> :   NetworkImage(
        'https://static-dev.collaction.org/profile-pictures/$_userId.png'),
      onForegroundImageError: (_, __) => const AssetImage(
        'assets/images/logo.png',
      ),
    );
  }
```
I think this looks better.

Reply:
I would still prefer handling `_userId == null` explicitly than hitting our backend with invalid requests.

Reply:
```suggestion
  Widget build(BuildContext context) {
    return CircleAvatar(
      maxRadius: _maxRadius,
      foregroundImage: 
        _image != null 
        ? FileImage(_image!) as ImageProvider<Object> 
        :  _userId != null 
          ?  NetworkImage('https://static-dev.collaction.org/profile-pictures/$_userId.png')
          : const AssetImage('assets/images/logo.png'),
      onForegroundImageError: (_, __) => const AssetImage('assets/images/logo.png'),
    );
  }
```